components/header/wallet_modal.tsx

```diff
@@ -0,0 +1,125 @@
+import { Dispatch, FormEvent, SetStateAction, useState } from 'react';
+import Image from 'next/image';
+import ReactModal from 'react-modal';
+
+import { validateWalletAddress } from '@utils';
+
+import arconnectLogo from '../../public/arconnect.png';
+import {
+	CloseModalButton,
+	ImportWalletErrorMessage,
+	ImportWalletForm,
+	ImportWalletInput,
+	ModalButton,
+	ModalButtons,
+	ModalContent,
+	ModalText,
+} from './wallet_modal.style';
+
+const WalletModal = ({
+	isOpen,
+	closeModal,
+	setWalletAddress,
+}: {
+	isOpen: boolean;
+	closeModal: Function;
+	setWalletAddress: Dispatch<SetStateAction<string>>;
+}): JSX.Element => {
+	const [isArconnectAvailable, setIsArconnectAvailable] = useState(false);
+	const [showWalletInput, setShowWalletInput] = useState(false);
+	const [importedPublicWalletAddress, setImportedPublicWalletAddress] = useState('');
+	const [importedPublicWalletAddressError, setImportedPublicWalletAddressError] = useState('');
+
+	if (typeof window !== 'undefined') {
+		window.addEventListener('arweaveWalletLoaded', async () => {
+			setIsArconnectAvailable(Boolean(window.arweaveWallet));
+
+			try {
+				const walletAddress = await window.arweaveWallet.getActiveAddress();
+				setWalletAddress(walletAddress);
+			} catch (e) {
+				console.log(e);
+			}
+		});
+	}
+
+	const connectToArconnect = async () => {
+		await window.arweaveWallet.connect(['ACCESS_ADDRESS'], { name: 'Inferno Rewards Leaderboard' });
+		const walletAddress = await window.arweaveWallet.getActiveAddress();
+
+		setWalletAddress(walletAddress);
+		closeModal();
+	};
+
+	const onImportWalletSubmit = (e: FormEvent<HTMLFormElement>) => {
+		const data = new FormData(e.target as HTMLFormElement);
```

Comment:
This casting appears to be unnecessary in my environment because of the type being declared above:

```suggestion
		const data = new FormData(e.target);
```

Reply:
You could consider typing it directly for what you need:

```ts
	const onImportWalletSubmit = (e: { target: HTMLFormElement }) => {
```

Reply:
That seemed to also cause type errors up stream for me FWIW

Reply:
I couldn't find any better alternative :(